diff -urN curl-7.86.0.orig/config.sub curl-7.86.0/config.sub
--- curl-7.86.0.orig/config.sub	2022-10-06 10:20:20.000000000 +0100
+++ curl-7.86.0/config.sub	2022-11-03 08:29:18.321600760 +0000
@@ -1754,7 +1754,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | kiwi*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -urN curl-7.86.0.orig/configure curl-7.86.0/configure
--- curl-7.86.0.orig/configure	2022-10-17 07:16:53.000000000 +0100
+++ curl-7.86.0/configure	2022-11-03 08:29:18.325600754 +0000
@@ -9287,7 +9287,7 @@
   fi
   ;;
 
-haiku*)
+haiku* | kiwi*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -15206,6 +15206,18 @@
   hardcode_into_libs=yes
   ;;
 
+kiwi*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  dynamic_linker="$host_os runtime_loader"
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  sys_lib_dlsearch_path_spec='/system/lib'
+  ;;
+
 hpux9* | hpux10* | hpux11*)
   # Give a soname corresponding to the major version so that dld.sl refuses to
   # link against other versions.
diff -urN curl-7.86.0.orig/include/curl/curl.h curl-7.86.0/include/curl/curl.h
--- curl-7.86.0.orig/include/curl/curl.h	2022-10-06 10:19:59.000000000 +0100
+++ curl-7.86.0/include/curl/curl.h	2022-11-03 08:29:18.325600754 +0000
@@ -76,7 +76,7 @@
     defined(__CYGWIN__) || defined(AMIGA) || defined(__NuttX__) || \
    (defined(__FreeBSD_version) && (__FreeBSD_version < 800000)) || \
    (defined(__MidnightBSD_version) && (__MidnightBSD_version < 100000)) || \
-    defined(__sun__)
+    defined(__sun__) || defined(__Kiwi__)
 #include <sys/select.h>
 #endif
 
diff -urN curl-7.86.0.orig/ltmain.sh curl-7.86.0/ltmain.sh
--- curl-7.86.0.orig/ltmain.sh	2022-10-06 10:20:15.000000000 +0100
+++ curl-7.86.0/ltmain.sh	2022-11-03 08:29:18.326600752 +0000
@@ -7283,7 +7283,7 @@
 	    # These systems don't actually have a C or math library (as such)
 	    continue
 	    ;;
-	  *-*-os2*)
+	  *-*-os2* | *-*-kiwi*)
 	    # These systems don't actually have a C library (as such)
 	    test X-lc = "X$arg" && continue
 	    ;;
@@ -9410,7 +9410,7 @@
       if test yes = "$build_libtool_libs"; then
 	if test -n "$rpath"; then
 	  case $host in
-	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2* | *-*-beos* | *-cegcc* | *-*-haiku*)
+	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2* | *-*-beos* | *-cegcc* | *-*-haiku* | *-*-kiwi*)
 	    # these systems don't actually have a c library (as such)!
 	    ;;
 	  *-*-rhapsody* | *-*-darwin1.[012])
diff -urN curl-7.86.0.orig/m4/libtool.m4 curl-7.86.0/m4/libtool.m4
--- curl-7.86.0.orig/m4/libtool.m4	2022-10-06 10:20:15.000000000 +0100
+++ curl-7.86.0/m4/libtool.m4	2022-11-03 08:29:18.327600751 +0000
@@ -2732,6 +2732,18 @@
   hardcode_into_libs=yes
   ;;
 
+kiwi*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  dynamic_linker="$host_os runtime_loader"
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  sys_lib_dlsearch_path_spec='/system/lib'
+  ;;
+
 hpux9* | hpux10* | hpux11*)
   # Give a soname corresponding to the major version so that dld.sl refuses to
   # link against other versions.
@@ -3535,7 +3547,7 @@
   fi
   ;;
 
-haiku*)
+haiku* | kiwi*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
