diff -urN bash-5.2.orig/builtins/ulimit.def bash-5.2/builtins/ulimit.def
--- bash-5.2.orig/builtins/ulimit.def	2022-10-16 08:40:13.641575178 +0100
+++ bash-5.2/builtins/ulimit.def	2022-10-16 08:44:17.743073205 +0100
@@ -23,6 +23,7 @@
 $BUILTIN ulimit
 $FUNCTION ulimit_builtin
 $DEPENDS_ON !_MINIX
+$DEPENDS_ON !__Kiwi__
 $SHORT_DOC ulimit [-SHabcdefiklmnpqrstuvxPRT] [limit]
 Modify shell resource limits.
 
@@ -71,7 +72,7 @@
 Returns success unless an invalid option is supplied or an error occurs.
 $END
 
-#if !defined (_MINIX)
+#if !defined (_MINIX) && !defined (__Kiwi__)
 
 #include <config.h>
 
@@ -806,4 +807,4 @@
   return retval;
 }
 
-#endif /* !_MINIX */
+#endif /* !_MINIX && !__Kiwi__ */
diff -urN bash-5.2.orig/configure bash-5.2/configure
--- bash-5.2.orig/configure	2022-10-16 08:40:13.593575251 +0100
+++ bash-5.2/configure	2022-10-16 08:44:17.745073201 +0100
@@ -21736,6 +21736,7 @@
 cygwin*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 opennt*|interix*) LOCAL_CFLAGS="-DNO_MAIN_ENV_ARG -DBROKEN_DIRENT_D_INO -D_POSIX_SOURCE -D_ALL_SOURCE -DRECYCLES_PIDS" ;;
 *openstep*)	LOCAL_CFLAGS="-D__APPLE_CC__" ;;
+kiwi*)	    LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 esac
 
 case "${host_os}-${CC}" in
diff -urN bash-5.2.orig/configure.ac bash-5.2/configure.ac
--- bash-5.2.orig/configure.ac	2022-10-16 08:40:13.614575219 +0100
+++ bash-5.2/configure.ac	2022-10-16 08:44:17.746073199 +0100
@@ -1183,6 +1183,7 @@
 cygwin*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 opennt*|interix*) LOCAL_CFLAGS="-DNO_MAIN_ENV_ARG -DBROKEN_DIRENT_D_INO -D_POSIX_SOURCE -D_ALL_SOURCE -DRECYCLES_PIDS" ;;
 *openstep*)	LOCAL_CFLAGS="-D__APPLE_CC__" ;;
+kiwi*)	    LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 esac
 
 dnl Stanza for OS/compiler pair-specific flags
diff -urN bash-5.2.orig/execute_cmd.c bash-5.2/execute_cmd.c
--- bash-5.2.orig/execute_cmd.c	2022-10-16 08:40:13.591575254 +0100
+++ bash-5.2/execute_cmd.c	2022-10-16 08:44:17.746073199 +0100
@@ -1384,8 +1384,10 @@
       selfb.ru_utime.tv_usec = kidsb.ru_utime.tv_usec = selfb.ru_stime.tv_usec = kidsb.ru_stime.tv_usec = 0;
       before = shellstart;
 #else
+#  if defined (HAVE_TIMES)
       before.tms_utime = before.tms_stime = before.tms_cutime = before.tms_cstime = 0;
       tbefore = shell_start_time;
+#  endif
 #endif
     }
 
diff -urN bash-5.2.orig/include/timer.h bash-5.2/include/timer.h
--- bash-5.2.orig/include/timer.h	2022-10-16 08:40:13.615575218 +0100
+++ bash-5.2/include/timer.h	2022-10-16 08:44:17.746073199 +0100
@@ -22,6 +22,8 @@
 #include "config.h"
 #endif
 
+#include <sys/time.h>
+
 #include "bashjmp.h"
 typedef struct _shtimer
 {
diff -urN bash-5.2.orig/lib/readline/terminal.c bash-5.2/lib/readline/terminal.c
--- bash-5.2.orig/lib/readline/terminal.c	2022-10-16 08:40:13.583575267 +0100
+++ bash-5.2/lib/readline/terminal.c	2022-10-16 08:44:17.747073197 +0100
@@ -102,7 +102,7 @@
 
 static int tcap_initialized;
 
-#if !defined (__linux__) && !defined (NCURSES_VERSION)
+#if !defined (__linux__) && !defined (NCURSES_VERSION) && !defined(__Kiwi__)
 #  if defined (__EMX__) || defined (NEED_EXTERN_PC)
 extern 
 #  endif /* __EMX__ || NEED_EXTERN_PC */
diff -urN bash-5.2.orig/lib/sh/random.c bash-5.2/lib/sh/random.c
--- bash-5.2.orig/lib/sh/random.c	2022-10-16 08:40:13.588575259 +0100
+++ bash-5.2/lib/sh/random.c	2022-10-16 08:44:17.747073197 +0100
@@ -22,6 +22,8 @@
 
 #include "bashtypes.h"
 
+#include <sys/time.h>
+
 #if defined (HAVE_SYS_RANDOM_H)
 #  include <sys/random.h>
 #endif
diff -urN bash-5.2.orig/parse.y bash-5.2/parse.y
--- bash-5.2.orig/parse.y	2022-10-16 08:40:13.623575205 +0100
+++ bash-5.2/parse.y	2022-10-16 08:44:17.747073197 +0100
@@ -2625,6 +2625,7 @@
       parser_state |= PST_ENDALIAS;
       /* We need to do this to make sure last_shell_getc_is_singlebyte returns
 	 true, since we are returning a single-byte space. */
+#if defined (HANDLE_MULTIBYTE)
       if (shell_input_line_index == shell_input_line_len && last_shell_getc_is_singlebyte == 0)
 	{
 #if 0
@@ -2638,6 +2639,7 @@
 	  shell_input_line_property[shell_input_line_index - 1] = 1;
 #endif
 	}
+#endif
       return ' ';	/* END_ALIAS */
     }
 #endif
diff -urN bash-5.2.orig/support/config.sub bash-5.2/support/config.sub
--- bash-5.2.orig/support/config.sub	2022-10-16 08:40:13.591575254 +0100
+++ bash-5.2/support/config.sub	2022-10-16 08:44:17.750073191 +0100
@@ -1754,7 +1754,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | kiwi*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -urN bash-5.2.orig/y.tab.c bash-5.2/y.tab.c
--- bash-5.2.orig/y.tab.c	2022-10-16 08:40:13.639575181 +0100
+++ bash-5.2/y.tab.c	2022-10-16 08:44:17.751073190 +0100
@@ -4936,6 +4936,7 @@
       parser_state |= PST_ENDALIAS;
       /* We need to do this to make sure last_shell_getc_is_singlebyte returns
 	 true, since we are returning a single-byte space. */
+#if defined (HANDLE_MULTIBYTE)
       if (shell_input_line_index == shell_input_line_len && last_shell_getc_is_singlebyte == 0)
 	{
 #if 0
@@ -4949,6 +4950,7 @@
 	  shell_input_line_property[shell_input_line_index - 1] = 1;
 #endif
 	}
+#endif
       return ' ';	/* END_ALIAS */
     }
 #endif
